### ISSUE TITLE
Angular 9.1.5 strange behavior

### ISSUE THREAD

**Issue by mlc-mlapis** [angular/angular#37000]
# 🐞 bug report

### Is this a regression?
✍️ Yes, the previous version in which this bug was not present was: 9.1.4

### Description
✍️ The standard compilation of previously long time valid code leads to a strange problem with a message like:
```
ERROR in Unable to write a reference to HeaderPanelComponent in .../github/m-eso/libs/shared/modules/core/src/lib/components/header-panel/header-panel.component.ts from .../github/m-eso/libs/shared/modules/core/src/lib/core.module.ts
```
The mentioned component and module are correct and contain valid code. It looks that the message is misleading and the base of the problem is some new change in the core of Angular 9.1.5.

## 🔬 Minimal Reproduction
Not sure if it is possible to reproduce because of the strange circumstances in a new version of Angular 9.1.5.

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.1.4
Node: 12.13.1
OS: win32 x64

Angular: 9.1.4
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect: 0.901.4
@angular-devkit/architect-cli: 0.901.4
@angular-devkit/build-angular: 0.901.4
@angular-devkit/build-optimizer: 0.901.4
@angular-devkit/build-webpack: 0.901.4
@angular-devkit/core: 9.1.4
@angular-devkit/schematics: 9.1.4
@angular-devkit/schematics-cli: 0.901.4
@angular/cdk: 9.2.2
@ngtools/webpack: 9.1.4
@schematics/angular: 9.1.4
@schematics/schematics: 0.901.4
@schematics/update: 0.901.4
rxjs: 6.5.5
typescript: 3.8.3
webpack: 4.42.0
</code></pre>

```
info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\NodeJS\\node.exe',
1 verbose cli   'C:\\NodeJS\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'stage'
1 verbose cli ]
2 info using npm@6.12.1
3 info using node@v12.13.1
4 verbose run-script [ 'prestage', 'stage', 'poststage' ]
5 info lifecycle m-eso@0.0.0~prestage: m-eso@0.0.0
6 info lifecycle m-eso@0.0.0~stage: m-eso@0.0.0
7 verbose lifecycle m-eso@0.0.0~stage: unsafe-perm in lifecycle true
...
9 verbose lifecycle m-eso@0.0.0~stage: CWD: C:\Data\WAMP\github\m-eso
10 silly lifecycle m-eso@0.0.0~stage: Args: [ '/d /s /c', 'ng build --configuration=staging' ]
11 silly lifecycle m-eso@0.0.0~stage: Returned: code: 1  signal: null
12 info lifecycle m-eso@0.0.0~stage: Failed to exec stage script
13 verbose stack Error: m-eso@0.0.0 stage: `ng build --configuration=staging`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Utils\NVM\v12.13.1\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:210:5)
13 verbose stack     at ChildProcess.<anonymous> (C:\Utils\NVM\v12.13.1\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:210:5)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
14 verbose pkgid m-eso@0.0.0
15 verbose cwd C:\Data\WAMP\github\m-eso
16 verbose Windows_NT 10.0.18363
17 verbose argv "C:\\NodeJS\\node.exe" "C:\\NodeJS\\node_modules\\npm\\bin\\npm-cli.js" "run" "stage"
18 verbose node v12.13.1
19 verbose npm  v6.12.1
20 error code ELIFECYCLE
21 error errno 1
22 error m-eso@0.0.0 stage: `ng build --configuration=staging`
22 error Exit status 1
23 error Failed at the m-eso@0.0.0 stage script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by petebacondarwin** [comment]
Probably a duplicate of #36992

**Comment by mlc-mlapis** [comment]
@petebacondarwin It could be but from our point of view, it is unclear what happened. 😸

**Comment by petebacondarwin** [comment]
We are investigating...

**Comment by petebacondarwin** [comment]
I am pretty confident that your scenario will be fixed by #37003 and/or #37008.
If you are able to try your project with the build artifacts from #37012, which can be found at https://circleci.com/gh/angular/angular/704553 and confirm whether this fixes your problem that would be very helpful.

You can find out more about how to do this here https://github.com/angular/angular/blob/master/docs/DEVELOPER.md#getting-packages-from-build-artifacts

**Comment by petebacondarwin** [comment]
We have just released 9.1.6 which should not have this problem. Please test.

**Comment by mlc-mlapis** [comment]
@petebacondarwin I did all tests and it really seems that with Angular 9.1.6 all is working correctly. Thank you. 👍
